engine/src/main/java/org/entando/entando/web/common/interceptor/EntandoOauth2Interceptor.java

```diff
@@ -56,54 +56,62 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
-                this.extractOAuthParameters(request, permission);
+
+                this.checkAuthorization(user, rqm.permission(), request);
             }
         }
         return true;
     }
 
-    protected void extractOAuthParameters(HttpServletRequest request, String permission) {
+    protected UserDetails extractOAuthParameters(HttpServletRequest request) {
         try {
-            logger.debug("Permission required: {}", permission);
+            request.getSession().setAttribute("user", null); //Clear previous session
+
             String accessToken = new EntandoBearerTokenExtractor().extractToken(request);
             if (StringUtils.isBlank(accessToken)) {
-                logger.warn("no access token found");
-                throw new EntandoTokenException("no access token found", request, null);
+                return null;
             }
+
             final OAuth2AccessToken token = this.getoAuth2TokenManager().readAccessToken(accessToken);
             this.validateToken(request, accessToken, token);
-            String username = null;
+            String username;
             if (token instanceof OAuth2AccessTokenImpl) {
                 username = ((OAuth2AccessTokenImpl) token).getLocalUser();
             } else {
                 Authentication auth = new EntandoBearerTokenExtractor().extract(request);
                 username = auth.getPrincipal().toString();
             }
-            this.checkAuthorization(username, permission, request);
+
+            UserDetails user = this.getAuthenticationProviderManager().getUser(username);
+            if (user == null) {
+                logger.warn("User {} not found ", username);
+                return null;
+            }
+
+            request.getSession().setAttribute("user", user);
+            return user;
         } catch (ApsSystemException ex) {
             logger.error("System exception {}", ex.getMessage());
             throw new EntandoTokenException("error parsing OAuth parameters", request, "guest");
         }
     }
 
-    protected void checkAuthorization(String username, String permission, HttpServletRequest request) throws ApsSystemException {
-        UserDetails user = this.getAuthenticationProviderManager().getUser(username);
-        if (user != null) {
-            request.getSession().setAttribute("user", user);
-            logger.debug("User {} requesting resource that requires {} permission ", username, permission);
-            if (StringUtils.isNotBlank(permission)) {
-                if (!this.getAuthorizationManager().isAuthOnPermission(user, permission)) {
-                    logger.warn("User {} is missing the required permission {}", username, permission);
-                    throw new EntandoAuthorizationException(null, request, username);
-                }
+    protected void checkAuthorization(UserDetails user, String permission, HttpServletRequest request) throws ApsSystemException {
+        if (null == user) {
```

Comment:
This is a pretty big change (even if an improvement). Is there a reason to change this flow to throw an exception when no user is present on the request? The original logged info and carried on.

Reply:
This exception was already being thrown in the `extractOAuthParameters` method.

As I explained in the PR text, the main change on this PR is to separate responsabilities as both `extractOAuthParameters` and `checkPermission` were throwing exceptions.

So by separating both methods, one's responsability is to just extract the token and user if present and the second is to actually validate it only if the annotation is configured to require a permission (and hence needs an authenticated user).

Check the method `preHandle`, as you can see, first the user is extracted and only if the annotation/permission is configured that we call the `checkPermission` method to guarantee that there is a logged in user and it has the correct permissions.